saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationFilter.java

```diff
@@ -86,7 +97,16 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 				localSpEntityId,
 				rp.getCredentials()
 		);
-		return getAuthenticationManager().authenticate(authentication);
+		return getAuthenticationManager(request).authenticate(authentication);
```

Comment:
It makes me a little nervous that an instance of this filter could contain both an `AuthenticationManager` and an `AuthenticationManagerResolver` instance. It could be a source of bugs in the future.

Have you already considered pushing the `AuthenticationManagerResolver` instead up to the parent? You could possibly have the parent hold onto only an instance of `AuthenticationManagerResolver` and have the `getAuthenticationManager(HttpServletRequest) method`.